lib/FastANI/utils/downloader.py

```diff
@@ -0,0 +1,47 @@
+"""
+Download either a Genome or Assembly.
+"""
+from DataFileUtil.DataFileUtilClient import DataFileUtil
+from AssemblyUtil.AssemblyUtilClient import AssemblyUtil
+
+
+def download_fasta(refs, cb_url):
+    """
+    Args:
+      ref - workspace reference in the form 'workspace_id/object_id/obj_version'
+      cb_url - callback server URL
+    Returns the path of the downloaded fasta file
+    """
+    dfu = DataFileUtil(cb_url)
+    assembly_util = AssemblyUtil(cb_url)
+    ws_objects = dfu.get_objects({'object_refs': refs})
+    paths = []
+    for (obj, ref) in zip(ws_objects['data'], refs):
+        ws_type = obj['info'][2]
+        if 'KBaseGenomes.Genome' in ws_type:
+            assembly_ref = get_assembly_ref_from_genome(ref, obj)
+        elif 'KBaseGenomeAnnotations.Assembly' in ws_type:
+            assembly_ref = ref
+        else:
+            raise TypeError('Invalid type ' + ws_type + '. Must be an Assembly or Genome.')
+        path = assembly_util.get_assembly_as_fasta({'ref': assembly_ref})['path']
+        paths.append(path)
+    return paths
+
+
+def get_assembly_ref_from_genome(genome_ref, ws_obj):
```

Comment:
This is how I grab the assembly reference from a genome object. Maybe there is a simpler way?

Reply:
Not very efficient to download the ws_obj twice -- once to get the type, then again to find the fasta files buried inside.